spec/element_base_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'spec_helper'
+
+RSpec.describe Kitchen::ElementBase do
+
+  let(:element) do
+    book_containing(html:
+      one_chapter_with_one_page_containing(
+        <<~HTML
+          <div data-type="example" class="class1" id="div1">
+            <p>This is a paragraph.</p>
+          </div>
+        HTML
+      )
+    )
+  end
+
+  describe '#has_class?' do
+    it 'returns true if element has given class' do
+      div_element = element.search('div[data-type="example"]').first
+      expect(div_element.has_class?('class1')).to eq true
+    end
+
+    it 'returns false if element does not have given class' do
+      div_element = element.search('div[data-type="example"]').first
+      expect(div_element.has_class?('class2')).to eq false
+    end
+  end
+
+  describe '#id' do
+    it 'returns the element\'s ID' do
+      div_element = element.search('div[data-type="example"]').first
+      expect(div_element.id).to eq 'div1'
+    end
+  end
+
+  describe '#id=' do
+    it 'sets the element\'s ID' do
+      para = element.search('p').first
+      para.id = 'para1'
+      expect(para.id).to match('para1')
```

Comment:
Use `.to eq` instead of `.to match` because `match` will return true if there's a partial match, e.g.

```
irb(main):001:0> "sectionblah".match("section")
=> #<MatchData "section">
```

Ditto below.